pandas/tests/groupby/test_categorical.py

```diff
@@ -860,3 +860,15 @@ def test_groupby_multiindex_categorical_datetime():
     expected = pd.DataFrame(
         {'values': [0, 4, 8, 3, 4, 5, 6, np.nan, 2]}, index=idx)
     assert_frame_equal(result, expected)
+
+
+def test_groupby_agg_observed_true_single_column():
+    # GH-23970
+    expected = pd.DataFrame({
+        'a': pd.Series([1, 1, 2], dtype='category'),
+        'b': [1, 2, 2], 'x': [1, 2, 3]})
+
+    result = expected.groupby(['a', 'b'],
```

Comment:
Might be reading this wrong but do we need the non-categorical column in the grouper to emulate this behavior?

Reply:
Yes. This is because the bug generates a group that doesn't exist due to using some weird Cartesian product of the two columns used in the grouper.

Therefore, while the only valid groupings are `(1, 1) (1, 2) (2, 2)`; the bug uses the CP of the two `set(1, 2) * set(1, 2)` which results in an additional non-existent `(2, 1)` grouping as well which is represented as `NaN` in the bugged output.

Reply:
Ok thanks for the responses. In that case I think you can improve the whatsnew to say that the observed keyword was essentially being ignored when selecting one column as part of the GroupBy. It’s currently too vague IMO

Reply:
Also used to be the case. But I changed it to be simpler on @mroeschke 's advice (is a previous resolved convo).

I can change it back if needed but would just like a confirmation.

Reply:
Clarity is always welcome. But generally whatsnew entries should refer to externally facing methods and implications.

Reply:
Yea the previous entry was arguably *too specific* mentioning the exact internal method where the problem arose. The current entry is too vague to be useful.

If you can meet somewhere in the middle on that I think it again would be mentioning that the `observed` keyword was not being respected, but without mentioning the private internal methods where the implementation was actually changed

Reply:
Pushed it. Hopefully this is a better middle ground between the 2 previous approaches!